File: src/api/project.ts
import { invoke } from '@tauri-apps/api/core'
import type { ProjectRow } from '../types/project'

export async function createProject(title: string, projectType: string): Promise<ProjectRow> {
  return invoke('create_project', { title, projectType })
}

export async function listProjects(): Promise<ProjectRow[]> {
  return invoke('list_projects')
}

export async function updateProject(id: string, title: string): Promise<ProjectRow> {
  return invoke('update_project', { id, title })
}

export async function deleteProject(id: string): Promise<void> {
  return invoke('delete_project', { id })
}

File: src-tauri/src/commands/project.rs
use crate::db::ProjectRow;
use crate::services::project::ProjectService;
use crate::utils::error::AppError;

#[tauri::command]
pub async fn create_project(title: String, project_type: String) -> Result<ProjectRow, AppError> {
    ProjectService::create(title, project_type).await
}

#[tauri::command]
pub async fn list_projects() -> Result<Vec<ProjectRow>, AppError> {
    ProjectService::list().await
}

#[tauri::command]
pub async fn update_project(id: String, title: String) -> Result<ProjectRow, AppError> {
    ProjectService::update(id, title).await
}

#[tauri::command]
pub async fn delete_project(id: String) -> Result<(), AppError> {
    ProjectService::delete(id).await
}
